Webpage and ReadMe update for 0.10.0 Snapshot (#1514)
arrow-kt/arrow: modules/docs/arrow-docs/docs/docs/quickstart/libraries/README.md

### arrow-optics

```groovy
dependencies {
    compile "io.arrow-kt:arrow-optics:$arrow_version"
}
```

Optics is the functional way of handling immutable data and collections in a way that's boilerplate free and efficient.

> It can be used alongside annotation processing to generate [simple DSLs]({{ '/docs/optics/dsl/' | relative_url }}) that read like imperative code.

Reviewer: "imperative code" isn't accurate since there is no alternative. Imperative code doesn't allow for complex nested traversal updates of mutable or immutable structures.

Not sure what to suggest. Perhaps something about what it does instead of what it looks like?

Author: What do you have in mind,  I solely merged the text description fro, the older version

Author: If you prefer we can also delete the description. And leave it as a dependency import section, maybe link a good example what optics can do

Reviewer: Nono, we should definitely have a good short description here to promote libraries.

Reviewer: ```suggestion
Arrow Optics offers a way of declaring how to focus deeply into immutable structure without boilerplate. It also offers an [Optics DSL]({{ '/docs/optics/dsl/' | relative_url }})  to elegantly describe complex use-cases in an elegant and simple manner without requiring to understand the underlying theory.
```

For all the new typeclasses it also includes the extensions available for basic types and datatypes in both arrow-core and arrow-extras.

Datatypes: [`Fold`]({{ '/docs/optics/fold/' | relative_url }}), [`Getter`]({{ '/docs/optics/getter/' | relative_url }}), [`Iso`]({{ '/docs/optics/iso/' | relative_url }}), [`Lens`]({{ '/docs/optics/lens/' | relative_url }}), [`Optional`]({{ '/docs/optics/optional/' | relative_url }}), [`Prism`]({{ '/docs/optics/prism/' | relative_url }}), [`Setter`]({{ '/docs/optics/setter/' | relative_url }}), [`Traversal`]({{ '/docs/optics/traversal/' | relative_url }})

Typeclasses: [`At`]({{ '/docs/optics/at/' | relative_url }}), [`Each`]({{ '/docs/optics/each/' | relative_url }}), [`FilterIndex`]({{ '/docs/optics/filterIndex/' | relative_url }}), [`Index`]({{ '/docs/optics/index/' | relative_url }})

### arrow-fx

```groovy
dependencies {
    compile "io.arrow-kt:arrow-fx:$arrow_version"
}
```

> The [fx library]({{ '/docs/effects/fx/' | relative_url }}) focuses on bridging suspend functions to typeclasses like Monad and Effect, and abstracts over concurrency frameworks using typeclasses.

Reviewer: > bridging suspend functions to typeclasses like Monad

This is not correct, you cannot call suspend functions from `Monad`. Also I'd like to elaborate a bit here to put some features in the spotlight. 

```suggestion
The [fx library]({{ '/docs/effects/fx/' | relative_url }}) offers a powerful concurrency DSL with an emphasis on easy concurrency and parallelism with guarantees about concurrent and parallel resource safety. It can be used with Arrow Fx's IO or a set of typeclasses to abstract over concurrency frameworks like `RxJava`, `Reactor`, `Arrow's IO`, etc.
```

### arrow-free

```groovy
dependencies {
    compile "io.arrow-kt:arrow-free:$arrow_version"
}
```

The [Free datatype]({{ '/docs/free/free/' | relative_url }}) is a way of interpreting domain specific languages from inside your program, including a configurable runner and flexible algebras.
This allows optimization of operations like operator fusion or parallelism, while remaining on your business domain.

Dependency: `arrow-free-data`

### arrow-free-data

```groovy
dependencies {
    compile "io.arrow-kt:arrow-free-data:$arrow_version"
}
```

This library includes the datatypes in Free. 

Datatypes: [`Free`]({{ '/docs/free/free/' | relative_url }}), [`FreeApplicative`]({{ '/docs/free/freeapplicative/' | relative_url }}), [`Cofree`]({{ '/docs/free/cofree/' | relative_url }}), [`Yoneda`]({{ '/docs/free/yoneda/' | relative_url }}), [`Coyoneda`]({{ '/docs/free/coyoneda/' | relative_url }})

Dependency: `arrow-core`

### arrow-aql

```groovy
dependencies {
    compile "io.arrow-kt:arrow-query-language:$arrow_version"
}
```

This [Arrow Query Library]({{ '/docs/aql/intro/' | relative_url }}) focuses on bringing SQL-like syntax to Arrow datatypes.

Dependency: `arrow-core`

> ## Annotation processors

Reviewer: I think this is awfully low on the page given that a lot of people ask for kapt setup, no? Should we perhaps add a small table of contents at the top so people can jump here?

Author: The more important thing is the core libraries everything else is still experimental and with the compiler plugins, some old libraries like arrow-syntax may as well disappear.
I only want that to be our focus even though we have other awesome projects.

These libraries focus on meta-programming to generate code that enables other libraries and constructs.

### arrow-meta

```groovy
dependencies {

>     compile "io.arrow-kt:arrow-meta:$arrow_version"

Reviewer: > Allows boilerplate generation for [`@extension`] instances and [`@higherkind`] datatypes.

This setup is wrong, the description belongs with the kapt setup.

}
```

Allows boilerplate generation for [`@extension`](({{ 'docs/patterns/glossary/#instances-and-extensions-interfaces' | relative_url }})) instances and [`@higherkind`]({{ 'https://arrow-kt.io/docs/patterns/glossary/#higher-kinds' | relative_url }}) datatypes.

### arrow-generic

```groovy
dependencies {

>     compile "io.arrow-kt:arrow-generic:$arrow_version"

Reviewer: Same here. Should be kapt setup.

Datatypes: [`Fold`]({{ '/docs/optics/fold/' | relative_url }}), [`Getter`]({{ '/docs/optics/getter/' | relative_url }}), [`Iso`]({{ '/docs/optics/iso/' | relative_url }}), [`Lens`]({{ '/docs/optics/lens/' | relative_url }}), [`Optional`]({{ '/docs/optics/optional/' | relative_url }}), [`Prism`]({{ '/docs/optics/prism/' | relative_url }}), [`Setter`]({{ '/docs/optics/setter/' | relative_url }}), [`Traversal`]({{ '/docs/optics/traversal/' | relative_url }})

Typeclasses: [`At`]({{ '/docs/optics/at/' | relative_url }}), [`Each`]({{ '/docs/optics/each/' | relative_url }}), [`FilterIndex`]({{ '/docs/optics/filterIndex/' | relative_url }}), [`Index`]({{ '/docs/optics/index/' | relative_url }})

### arrow-fx

```groovy
dependencies {
    compile "io.arrow-kt:arrow-fx:$arrow_version"
}
```

The [fx library]({{ '/docs/effects/fx/' | relative_url }}) offers a powerful concurrency DSL with an emphasis on easy concurrency and parallelism with guarantees about concurrent and parallel resource safety. It can be used with Arrow Fx's IO or a set of typeclasses to abstract over concurrency frameworks like `RxJava`, `Reactor`, `Arrow's IO`, etc.

> Additionally it provides its own concurrency primitive, called IO.

Reviewer: ```suggestion
```

This can be removed since it's already in the paragraph and in the datatypes below. All the concurrency primitives are also missing from this list.

{:.beginner}

> beginner

Reviewer 2: remove tags for beginner etc...

Author: also the tags for advanced etc.

```groovy
dependencies {
    compile "io.arrow-kt:arrow-core:$arrow_version"
}
```

> This library include the possible [typeclass extensions]({{ '/docs/patterns/glossary/#instances-and-extensions-interfaces' | relative_url }}) including datatypes and typeclasses from `arrow-core-data` that can be implemented for the datatypes in their respective libraries.

Reviewer 2: No need to mention any of this here, README should just include a way to get the main artifact loaded in a project. This can be explained in the advanced page

Author: this is the advanced page.
